packages/expo-updates/android/src/main/java/expo/modules/updates/selectionpolicy/LauncherSelectionPolicyOverrideAware.kt

```diff
@@ -0,0 +1,25 @@
+package expo.modules.updates.selectionpolicy
+
+import expo.modules.updates.db.entity.UpdateEntity
+import org.json.JSONObject
+
+/**
+ * A LauncherSelectionPolicy that prioritizes updates fetched with UpdatesConfigurationOverride.
+ * This works similarly to LauncherSelectionPolicySingleUpdate but targets updates from override URLs
+ * rather than a specific UUID.
+ */
+internal class LauncherSelectionPolicyOverrideAware : LauncherSelectionPolicy {
+  override fun selectUpdateToLaunch(
+    updates: List<UpdateEntity>,
+    filters: JSONObject?
+  ): UpdateEntity? {
+    val candidates = updates
+      .filter { SelectionPolicies.matchesFilters(it, filters) }
+      .sortedByDescending { it.commitTime }
+
+    val firstOverrideUpdate: UpdateEntity? = candidates.firstOrNull { it.isFromOverride }
```

Comment:
is there an edge case where the URL is overridden to the same URL as the original config? what should happen in that case?

Reply:
sounds like people call `setUpdateURLAndRequestHeadersOverride` and pass the updateUrl as the url in app.json? it should be fine to fetch updates from the same url and having `isFromOverride` in database. we should launch into the same update still

Reply:
Hard to wrap my head around it, but the concern is that:
1. User uses app normally for URL `blah.com/update`, downloads update with ID 123.
2. Override is done, but URL is set to the same thing: `blah.com/update`... which means that the update downloaded will have the same ID (123). I assume this means it won't be added to the database (it's already there).
3. So, when it comes time to select the update, it won't have the override field and null update will be returned. (it should return the latest update)

Clearing the build data also wouldn't solve this since the URL and headers would be the same (it's consistent). 

Trying to think about how to fix this is tough. Do we update the row set is_override? Do we clear updates from the db when override is called?